velox/exec/HashBuild.cpp

```diff
@@ -1117,13 +1121,22 @@ void HashBuild::reclaim(
                 : (spiller_->finalized() ? "finalized" : "non-finalized"))
         << "] " << pool()->name()
         << ", usage: " << succinctBytes(pool()->usedBytes());
+
+    // Worst case scenario reservation was performed in ensureTableFits() when
```

Comment:
Can we unconditionally release the reserved memory even before the max spill limit check? Thanks!